Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide.hs

```diff
@@ -379,7 +380,12 @@ endRide handle@ServiceHandle {..} rideId req = withLogTag ("rideId-" <> rideId.g
                         else
                           if (isJust updRide.estimatedTollCharges || isJust updRide.tollCharges) && distanceCalculationFailure
                             then (Nothing, Nothing, Just Unsure)
-                            else (updRide.tollCharges, updRide.tollNames, Just Sure)
+                            else
+                              if (isJust updRide.estimatedTollCharges) && (updRide.tollConfidence == Just Unsure || updRide.tollConfidence == Nothing) && (isJust updRide.driverDeviatedToTollRoute)
```

Comment:
@RahulGupta0706 updRide would have tollConfidence as Nothing at this point during comparison since it get's updated on this line https://github.com/nammayatri/nammayatri/pull/9206/files#diff-2584aff629f8495cba7c62021201da759c8cf3b9fdff7f91010910bc82dac2d8L384

You can modify your logic as below:
```
let (tollCharges, tollNames, tollConfidence) = do
      let distanceCalculationFailure = distanceCalculationFailed || (maybe False (> 0) updRide.numberOfSelfTuned)
      if distanceCalculationFailure
        then 
          if isJust updRide.estimatedTollCharges
            then 
              if isJust updRide.tollCharges
                then (updRide.tollCharges, updRide.tollNames, Just Neutral)
                else (updRide.estimatedTollCharges, updRide.estimatedTollNames, Just Neutral)
            else (Nothing, Nothing, Just Unsure)
        else (updRide.tollCharges, updRide.tollNames, Just Sure)
```